### PR TITLE
[ti-ctl]:add start_ts as filter in scan & mvcc

### PR DESCRIPTION
**Pull request by AndreMouche**
@disksing @BusyJay PTAL

### CODE REVIEW

**Comment by BusyJay** [comment]
@zhangjinpeng1987 PTAL

File: src/bin/tikv-ctl.rs [hunk 1]
```diff
@@ -147,31 +155,33 @@ fn main() {
...
         let key_encoded = matches.is_present("encoded");
+        let start_ts = matches.value_of("start_ts").map(|s| s.parse().unwrap());
         println!("You are searching Key {}: ", key);
         match cf_name {
             CF_DEFAULT => {
-                dump_mvcc_default(&db, key, key_encoded);
+                dump_mvcc_default(&db, key, key_encoded, start_ts);
             }
             CF_LOCK => {
                 dump_mvcc_lock(&db, key, key_encoded);
```
Comment (zhangjinpeng1987) [inline src/bin/tikv-ctl.rs]: how about lock?

**Comment by zhangjinpeng1987** [comment]
any test?

**Comment by AndreMouche** [comment]
It seems impossible to verify the result in this case? @zhangjinpeng1987

**Comment by siddontang** [comment]
I think we don't need `Option<u64>` for start/commit TS here, we can check 0 directly.

**Comment by siddontang** [comment]
Do we need to support checking in range `[ts1, ts2)` @zhangjinpeng1987

**Comment by zhangjinpeng1987** [comment]
No deed to support checking in range.

File: src/bin/tikv-ctl.rs [hunk 2]
```diff
@@ -337,26 +366,64 @@ fn dump_all_region_info(db: DB, skip_tombstone: bool) {
...
 
-fn dump_range(db: DB, from: String, to: Option<String>, limit: Option<u64>, cf: &str) {
+fn parse_ts_key_from_key(encode_key: Vec<u8>) -> (u64, String) {
+    let item_key = Key::from_encoded(encode_key);
+    let ts = item_key.decode_ts().unwrap();
+    let item_key = item_key.truncate_ts().unwrap();
+    let key_vec = item_key.encoded();
+
+    let key = String::from_utf8(key_vec.clone()).unwrap();
+    // println!("ts:{:?},key_vec:{:?}",ts,key.clone());
```
Comment (disksing) [inline src/bin/tikv-ctl.rs]: Remove redundant logs.

**Comment by siddontang** [comment]
PTAL @disksing @zhangjinpeng1987

File: src/bin/tikv-ctl.rs [hunk 3]
```diff
@@ -337,26 +366,64 @@ fn dump_all_region_info(db: DB, skip_tombstone: bool) {
         .unwrap();
 }
 
-fn dump_range(db: DB, from: String, to: Option<String>, limit: Option<u64>, cf: &str) {
+fn parse_ts_key_from_key(encode_key: Vec<u8>) -> (u64, String) {
```
Comment (disksing) [inline src/bin/tikv-ctl.rs]: I've noticed that the returned String is never used.

File: src/bin/tikv-ctl.rs [hunk 4]
```diff
@@ -337,26 +366,64 @@ fn dump_all_region_info(db: DB, skip_tombstone: bool) {
...
                  &mut |k, v| {
-                     println!("key: {}, value len: {}", escape(k), v.len());
-                     println!("{}", escape(v));
-                     cnt += 1;
-                     Ok(cnt < limit)
-                 })
+            let mut right_key = true;
+            match cf {
+                CF_DEFAULT => {
+                    let (ts, _) = parse_ts_key_from_key(escape(k).into_bytes());
```
Comment (disksing) [inline src/bin/tikv-ctl.rs]: I think `let ts = Key::from_encoded(k).decode_ts().unwrap()` will work well.

Comment (AndreMouche) [inline src/bin/tikv-ctl.rs]: It seems I need the method parse_ts_key_from_key in [issues_1284](https://github.com/pingcap/tikv/issues/1284) @disksing

**Comment by siddontang** [comment]
PTAL @disksing @hhkbp2

**Comment by disksing** [comment]
LGTM

**Comment by hhkbp2** [comment]
LGTM

**Pull request merged**